examples/CMSapp/CMSapp.cu

```diff
@@ -0,0 +1,452 @@
+// SPDX-FileCopyrightText: 2021 CERN
+// SPDX-License-Identifier: Apache-2.0
+
+#include "CMSapp.h"
+#include "CMSapp.cuh"
+
+#include <AdePT/Atomic.h>
+#include <AdePT/BVHNavigator.h>
+#include <AdePT/MParray.h>
+
+#include <CopCore/Global.h>
+#include <CopCore/PhysicalConstants.h>
+#include <CopCore/Ranluxpp.h>
+
+#include <VecGeom/base/Config.h>
+#include <VecGeom/base/Stopwatch.h>
+#include <VecGeom/management/GeoManager.h>
+#ifdef VECGEOM_ENABLE_CUDA
+#include <VecGeom/backend/cuda/Interface.h>
+#endif
+
+#include <G4HepEmState.hh>
+#include <G4HepEmData.hh>
+#include <G4HepEmElectronInit.hh>
+#include <G4HepEmGammaInit.hh>
+#include <G4HepEmMatCutData.hh>
+#include <G4HepEmMaterialInit.hh>
+#include <G4HepEmParameters.hh>
+#include <G4HepEmParametersInit.hh>
+
+#include <iostream>
+#include <iomanip>
+#include <stdio.h>
+
+__constant__ __device__ struct G4HepEmParameters g4HepEmPars;
+__constant__ __device__ struct G4HepEmData g4HepEmData;
+
+__constant__ __device__ int *MCIndex = nullptr;
+
+__constant__ __device__ int Zero = 0;
+
+//struct G4HepEmState {
+//  G4HepEmData data;
+//  G4HepEmParameters parameters;
+//};
+
+void InitG4HepEmGPU(G4HepEmState *state)
+{
+  // Copy to GPU.
+  CopyG4HepEmDataToGPU(state->fData);
+  COPCORE_CUDA_CHECK(cudaMemcpyToSymbol(g4HepEmPars, state->fParameters, sizeof(G4HepEmParameters)));
+
+  // Create G4HepEmData with the device pointers.
+  G4HepEmData dataOnDevice;
+  dataOnDevice.fTheMatCutData   = state->fData->fTheMatCutData_gpu;
+  dataOnDevice.fTheMaterialData = state->fData->fTheMaterialData_gpu;
+  dataOnDevice.fTheElementData  = state->fData->fTheElementData_gpu;
+  dataOnDevice.fTheElectronData = state->fData->fTheElectronData_gpu;
+  dataOnDevice.fThePositronData = state->fData->fThePositronData_gpu;
+  dataOnDevice.fTheSBTableData  = state->fData->fTheSBTableData_gpu;
+  dataOnDevice.fTheGammaData    = state->fData->fTheGammaData_gpu;
+  // The other pointers should never be used.
+  dataOnDevice.fTheMatCutData_gpu   = nullptr;
+  dataOnDevice.fTheMaterialData_gpu = nullptr;
+  dataOnDevice.fTheElementData_gpu  = nullptr;
+  dataOnDevice.fTheElectronData_gpu = nullptr;
+  dataOnDevice.fThePositronData_gpu = nullptr;
+  dataOnDevice.fTheSBTableData_gpu  = nullptr;
+  dataOnDevice.fTheGammaData_gpu    = nullptr;
+
+  COPCORE_CUDA_CHECK(cudaMemcpyToSymbol(g4HepEmData, &dataOnDevice, sizeof(G4HepEmData)));
+}
+
+static void FreeG4HepEm(G4HepEmState *state)
+{
+  FreeG4HepEmData(state->fData);
+}
```

Comment:
AFAICT G4HepEm is now initialized in `CMSapp.cpp`, so memory should also be freed there.